src/Compilers/Core/Portable/PEWriter/Members.cs

```diff
@@ -979,5 +979,26 @@ internal static bool HasBody(this IMethodDefinition methodDef)
             return !methodDef.IsAbstract && !methodDef.IsExternal &&
                 (methodDef.ContainingTypeDefinition == null || !methodDef.ContainingTypeDefinition.IsComObject);
         }
+
+        /// <summary>
+        /// When emitting ref assemblies, some members will not be included.
+        /// </summary>
+        public static bool ShouldInclude(this ITypeDefinitionMember member, EmitContext context)
+        {
+            var method = member as IMethodDefinition;
+            if (method != null && method.IsVirtual)
+            {
+                return true;
+            }
+
+            switch (member.Visibility)
+            {
+                case TypeMemberVisibility.Private:
+                    return context.IncludePrivateMembers;
+                case TypeMemberVisibility.Assembly:
```

Comment:
>case [](start = 16, length = 4)

`TypeMemberVisibility.FamilyAndAssembly` should be handled the same way as `Assembly`.